src/mpid/ch4/shm/ipc/gpu/gpu_post.c

```diff
@@ -8,14 +8,19 @@
 cvars:
     - name        : MPIR_CVAR_CH4_IPC_GPU_HANDLE_CACHE
       category    : CH4
-      type        : int
-      default     : 1
+      type        : enum
+      default     : specialized
       class       : none
       verbosity   : MPI_T_VERBOSITY_USER_BASIC
       scope       : MPI_T_SCOPE_ALL_EQ
-      description : >-
-        By default, we will cache ipc handle. To manually disable ipc
-        handle cache, user can set this variable to 0.
+      description : |-
+        By default, we will cache ipc handles using the specialized cache mechanism. If the
+        gpu-specific backend does not implement a specialized cache, then we will fallback to
+        the generic cache mechanism. Users can optionally force the generic cache mechanism or
+        disable ipc caching entirely.
+        generic - use the cache mechanism in the generic layer
+        specialized - use the cache mechanism in a gpu-specific mpl layer (if applicable)
+        disabled - disable caching completely
```

Comment:
Wrong commit title ("mpl/gpu: enable mpl-specific cache mechanisms"). I think it more fitting with "ch4/gpu: default MPIR_CVAR_CH4_IPC_GPU_HANDLE_CACHE to specialized"

Reply:
Fixed.

There will be no impact to non-ze path. Before calling `MPL_gpu_init`, we set the `specialized_cache` member of `MPL_gpu_info` based on the CVAR:

```
        bool specialized_cache =
            (MPIR_CVAR_CH4_IPC_GPU_HANDLE_CACHE == MPIR_CVAR_CH4_IPC_GPU_HANDLE_CACHE_specialized);

        MPL_gpu_info.specialized_cache = specialized_cache;

        int mpl_errno = MPL_gpu_init(debug_summary);
        MPIR_ERR_CHKANDJUMP(mpl_errno != MPL_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**gpu_init");
```

Then check the value after `MPL_gpu_init`. If it is `false` (which is set for fallback, CUDA, and HIP), then it falls back to generic cache (i.e. AVL cache):

```
            /* If the MPL gpu backend doesn't support specialized cache, fallback to generic. */
            if (specialized_cache && !MPL_gpu_info.specialized_cache) {
                MPIR_CVAR_CH4_IPC_GPU_HANDLE_CACHE = MPIR_CVAR_CH4_IPC_GPU_HANDLE_CACHE_generic;
            }
```